source/fundamentals/logging.txt

```diff
@@ -7,37 +7,173 @@ Logging
+level also logs all messages with higher severity levels. For example, setting
+the log level to ``critical`` also logs messages with severity levels of
+``emergency`` and ``alert``.
 
-Temporary Alternative
----------------------
+The lower the severity level you specify, the more information the driver logs,
+which impacts performance of your application. For the best performance, ensure
+that you set the severity to the highest severity level that provides the information
+your application needs.
 
-We are developing a new logging framework. In the meantime, you can output monitor events
-by using the following snippet:
+Configuration
+-------------
 
-.. code-block:: javascript
+You can configure logging in the {+driver-short+} without code changes by
+specifying environment variables. You can also configure logging
+programmatically by specifying client options in the ``MongoClient``
+constructor.
 
-   const uri = "mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority";
-   const client = new MongoClient(uri, { monitorCommands:true });
+.. note::
 
-   client.on('commandStarted', (event) => console.debug(event));
-   client.on('commandSucceeded', (event) => console.debug(event));
-   client.on('commandFailed', (event) => console.debug(event));
+   Because connection strings are often shared among a deployment of different
+   appications that might have different logging support, we do not recommend
+   using a connection string to configure logging.
+
+Environment Variables
+~~~~~~~~~~~~~~~~~~~~~
+
+You can configure logging for different components of the driver by specifying
+a severity level in one or more of the following environment variables:
+
+- ``MONGODB_LOG_COMMAND``
+- ``MONGODB_LOG_TOPOLOGY``
+- ``MONGODB_LOG_SERVER_SELECTION``
+- ``MONGODB_LOG_CONNECTION``
+- ``MONGODB_LOG_CLIENT``
+
+If you don't specify any of the preceding environment variables, the driver uses
+the value of ``MONGODB_LOG_ALL``, which if not specified, is implicitly set to
+``off``.
+
+.. tip::
+
+   Logging at the command level is the most performance-heavy logging option
+   due to the frequency in which commands are sent to the server. Only
+   specify this option if necessary for your application.
+
+The following example sets the log level for all components to ``debug`` except
+for ``MONGODB_LOG_COMMAND``:
+
+.. code-block:: bash
+
+   export MONGODB_LOG_ALL="debug"
+   export MONGODB_LOG_COMMAND="off"
+
+You can specify whether the driver logs to ``stderr`` or ``stdout`` by setting
+the ``MONGODB_LOG_PATH`` environment variable to ``"stderr"`` or ``"stdout"``.
+By default, the driver logs to ``stdout``.
+
+The driver stringifies logged documents by using EJSON, which limits strings to
+1000 characters by default. You can specify a maximum document length for your
+logger by specifying the ``MONGODB_LOG_MAX_DOCUMENT`` environment variable. Set
+this variable to ``0`` to perform no truncation.
```

Comment:
S: consider putting all of the logger options in a table, as this section is a bit disorganized at the moment